Library/DaylightModelController.swift

```diff
@@ -0,0 +1,70 @@
+import UIKit
+
+protocol DaylightModelControllerDelegate: class {
+    func daylightModelControllerDidUpdate(_ controller: DaylightModelController)
+}
+
+class DaylightModelController {
+    weak var delegate: DaylightModelControllerDelegate? {
+        didSet {
+            self.updateDelegate()
```

Comment:
This seems kind of circular - why does the delegate need to be updated when a new delegate is set?

Reply:
True! This is stupid, i should still look for the right time to update the delegates when the model is set... 

But it is something that hould happen, since the VC's or views are updated with a fresh controller object, so when you set the modelController the view also needs to be updated with that object!

Reply:
You already have to have a reference to the controller to set the delegate on it, right?